jumanji/environments/routing/lbf/env.py

```diff
@@ -0,0 +1,411 @@
+        """
+        state = self._generator(key)
+        observation = self._observer.state_to_observation(state)
+        timestep = restart(observation, shape=self.num_agents)
+        timestep.extras = self._get_extra_info(state, timestep)
+
+        return state, timestep
+
+    def step(self, state: State, actions: chex.Array) -> Tuple[State, TimeStep]:
+        """Simulate one step of the environment.
+
+        Args:
+            state (State): State  containing the dynamics of the environment.
+            actions (chex.Array): Array containing the actions to take for each agent.
+
+        Returns:
+            Tuple[State, TimeStep]: `State` object corresponding to the next state and
+            `TimeStep` object corresponding the timestep returned by the environment.
+        """
+        # Move agents, fix collisions that may happen and set loading status.
+        moved_agents = utils.update_agent_positions(
+            state.agents, actions, state.food_items, self.grid_size
+        )
+
+        # Eat the food
+        food_items, eaten_this_step, adj_loading_agents_levels = jax.vmap(
+            utils.eat_food, (None, 0)
+        )(moved_agents, state.food_items)
+
+        reward = self.get_reward(food_items, adj_loading_agents_levels, eaten_this_step)
+
+        state = State(
+            agents=moved_agents,
+            food_items=food_items,
+            step_count=state.step_count + 1,
+            key=state.key,
+        )
+        observation = self._observer.state_to_observation(state)
+
+        # First condition is truncation, second is termination.
+        terminate = jnp.all(state.food_items.eaten)
+        truncate = state.step_count >= self.time_limit
+
+        timestep = jax.lax.switch(
+            terminate + 2 * truncate,
+            [
+                # !terminate !trunc
+                lambda rew, obs: transition(
+                    reward=rew, observation=obs, shape=self.num_agents
+                ),
+                # terminate !truncate
+                lambda rew, obs: termination(
+                    reward=rew, observation=obs, shape=self.num_agents
+                ),
+                # !terminate truncate
+                lambda rew, obs: truncation(
+                    reward=rew, observation=obs, shape=self.num_agents
+                ),
+                # terminate truncate
+                lambda rew, obs: termination(
+                    reward=rew, observation=obs, shape=self.num_agents
+                ),
+            ],
+            reward,
+            observation,
+        )
+        timestep.extras = self._get_extra_info(state, timestep)
+
+        return state, timestep
+
+    def _get_extra_info(self, state: State, timestep: TimeStep) -> Dict:
+        """Computes extras metrics to be returned within the timestep."""
+        n_eaten = state.food_items.eaten.sum() + timestep.extras.get(
+            "eaten_food", jnp.float32(0)
+        )
```

Comment:
This should always be an int
```suggestion
        n_eaten = state.food_items.eaten.sum() + timestep.extras.get(
            "eaten_food", jnp.int32(0)
        )
```